src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/PrivateWorkerCraftingRequestResolver.java

```diff
@@ -0,0 +1,159 @@
+                }
+
+                return ((AbstractBuildingWorker) building).canCraft(stack);
+            }
+        }
+
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public List<IToken<?>> attemptResolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        if (manager.getColony().getWorld().isRemote)
+        {
+            return null;
+        }
+
+        final Colony colony = (Colony) manager.getColony();
+        final ILocation requesterLocation = request.getRequester().getRequesterLocation();
+        final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+        if(canResolve(manager, request) && building instanceof AbstractBuildingWorker)
+        {
+            final ItemStack stack = request.getRequest().getResult();
+            final IRecipeStorage storage = ((AbstractBuildingWorker) building).getFirstFullFillableRecipe(stack);
+            final List<IToken<?>> tokens = new ArrayList<>();
+            if(storage == null)
+            {
+                return null;
+            }
+
+            //todo After simulation has been added, we need to simulate the subrequest and decided wheter to try to resolve it or not.
+            for(final ItemStack neededStack: storage.getInput())
+            {
+                final Stack stackRequest = new Stack(neededStack);
+                tokens.add(manager.createRequest(this, stackRequest));
+            }
+
+            return tokens;
+        }
+
+        return null;
+    }
+
+    @Override
+    public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        final Colony colony = (Colony) manager.getColony();
+        final ILocation requesterLocation = request.getRequester().getRequesterLocation();
+        final AbstractBuilding building = colony.getBuilding(requesterLocation.getInDimensionLocation());
+        final ItemStack stack = request.getRequest().getResult();
+
+        final IRecipeStorage storage = ((AbstractBuildingWorker) building).getFirstFullFillableRecipe(stack);
+
+        if(storage == null)
+        {
+            return;
+        }
+
+        ((AbstractBuildingWorker) building).fullFillRecipe(storage);
+    }
+
+    @Nullable
+    @Override
+    public IRequest<?> getFollowupRequestForCompletion(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> completedRequest)
+    {
+        //No followup needed.
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public IRequest<?> onRequestCancelledOrOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends IDeliverable> request)
+    {
+        return null;
+    }
+
+    @Override
+    public void onRequestComplete(@NotNull final IToken<?> token)
+    {
+    }
+
+    @Override
+    public void onRequestCancelled(@NotNull final IToken<?> token)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)